Formula/cherrytree.rb

```diff
@@ -0,0 +1,51 @@
+class Cherrytree < Formula
+  desc "Hierarchical note taking application featuring rich text and syntax highlighting"
+  homepage "https://www.giuspen.com/cherrytree"
+  url "https://www.giuspen.com/software/cherrytree_0.99.30.tar.xz"
+  sha256 "dff54e8c484beb35531a9aa0987759bc3d7e426979473ada40d1d022fdc50120"
+  license "GPL-3.0-or-later"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "python@3.9" => :build
+  depends_on "adwaita-icon-theme"
+  depends_on "fmt"
+  depends_on "gspell"
+  depends_on "gtksourceviewmm3"
+  depends_on "libxml++"
+  depends_on "spdlog"
+  depends_on "uchardet"
+  uses_from_macos "curl"
+
+  def install
+    system "cmake", ".", "-DBUILD_TESTING=''", *std_cmake_args
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"homebrew.ctd").write <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+      <cherrytree>
+        <bookmarks list=""/>
+        <node name="rich text" unique_id="1" prog_lang="custom-colors" tags="" readonly="0" custom_icon_id="0" is_bold="0" foreground="" ts_creation="1611952177" ts_lastsave="1611952932">
+          <rich_text>this is a </rich_text>
+          <rich_text weight="heavy">simple</rich_text>
+          <rich_text> </rich_text>
+          <rich_text foreground="#ffff00000000">command line</rich_text>
+          <rich_text> </rich_text>
+          <rich_text style="italic">test</rich_text>
+          <rich_text> </rich_text>
+          <rich_text family="monospace">for</rich_text>
+          <rich_text> </rich_text>
+          <rich_text link="webs https://brew.sh/">homebrew</rich_text>
+        </node>
+        <node name="code" unique_id="2" prog_lang="python3" tags="" readonly="0" custom_icon_id="0" is_bold="0" foreground="" ts_creation="1611952391" ts_lastsave="1611952667">
+          <rich_text>print("hello world")</rich_text>
+        </node>
+      </cherrytree>
+    EOS
+    system "#{bin}/cherrytree", testpath/"homebrew.ctd", "--export_to_txt_dir", testpath, "--export_single_file"
+    assert_predicate testpath/"homebrew.ctd.txt", :exist?
```

Comment:
Is it worth checking the content of `homebrew.ctd.txt`? You can use `assert_match` to check a substring.

Reply:
Indeed I would like to check the existence of strings in the file rather than the exact whole file content, I'll give it a try with `assert_match` thanks